datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -160,6 +176,16 @@ impl ExecutionPlan for CoalesceBatchesExec {
     fn statistics(&self) -> Result<Statistics> {
         self.input.statistics()
     }
+
+    fn with_fetch(&self, limit: Option<usize>) -> Option<Arc<dyn ExecutionPlan>> {
```

Comment:
It might help to document somewhere (perhaps the struct's documentation) why setting `limit` is valuable on `CoalesceBatchesExec ` -- in this case I think it is because `CoalesceBatchesExec` buffers data so it may buffer more than necessary waiting for a limit to complete